File: hrms-backend/src/main/java/sep490/com/example/hrms_backend/service/RoleService.java
package sep490.com.example.hrms_backend.service;

import sep490.com.example.hrms_backend.dto.PermissionDTO;
import sep490.com.example.hrms_backend.dto.RoleDTO;

import java.util.List;
import java.util.Set;

public interface RoleService {
    void updatePermissions(Long roleId, Set<Long> permissionIds);
    Set<PermissionDTO> getPermissionsByRole(Long roleId);
    List<RoleDTO> getAllRoles();

}

File: hrms-frontend/src/services/roleService.js
import axiosClient from "./axiosClient";

const roleService = {
  getAllRoles: () => axiosClient.get("/roles"),
  getPermissionsByRole: (roleId) =>
    axiosClient.get(`/roles/${roleId}/permissions`),
  updatePermissions: (roleId, permissionIds) =>
    axiosClient.put(`/roles/${roleId}/permissions`, {
      permissionIds,
    }),
};

export default roleService;
